File: pivot.R
#!/usr/bin/env Rscript

# Load the necessary libraries
library(dplyr)

# Read the CSV file into a data frame
data <- read.csv('boston_311_2023_raw.csv', header = TRUE)

# Group the data by the 'reason' column and count the occurrences of each reason
reason_counts <- data %>%
  group_by(reason) %>%
  summarise(Count = n())

# Print the result (optional)
print(reason_counts)

# Save the result to a new CSV file
write.csv(reason_counts, file = 'boston_311_2023_by_reason.csv', row.names = FALSE)

File: pivot.py
#!/usr/bin/env python3

import pandas as pd

# Read the CSV file into a pandas DataFrame
data = pd.read_csv('boston_311_2025_raw.csv')

# Group the data by the 'reason' column and count the occurrences of each reason
reason_counts = data.groupby('reason').size().reset_index(name='Count')

# Save the result to a new CSV file
reason_counts.to_csv('boston_311_2025_by_reason.csv', index=False)

# Print the result (optional)
print(reason_counts)
